### PR TITLE
set max connections for passthrough cluster

### PR DESCRIPTION
**Pull request by dwradcliffe**
Fixes https://github.com/istio/istio/issues/13771

The value here is fairly arbitrary. I'm trying to set it very high so we never run into the max.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -982,6 +982,14 @@ func buildDefaultPassthroughCluster(env *model.Environment) *apiv2.Cluster {
 		ConnectTimeout:       util.GogoDurationToDuration(env.Mesh.ConnectTimeout),
 		LbPolicy:             apiv2.Cluster_ORIGINAL_DST_LB,
 	}
+	passthroughSettings := &networking.ConnectionPoolSettings{
+		Tcp: &networking.ConnectionPoolSettings_TCPSettings{
+			// The envoy default is 1024. This isn't configurable right now so we set
+			// this to a very high value so outbound connections are not limited.
+			MaxConnections: 1024 * 100,
+		},
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: is this all you want? there are other settings in the http piece, like max requests per connection, max retries, etc.. I would suggest bumping them up as well.

**Comment by linsun** [comment]
@dwradcliffe could you also generate a PR for 1.2 release branch? 

cc @fpesce

**Pull request merged**